src/confcom/azext_confcom/security_policy.py

```diff
@@ -935,19 +935,35 @@ def load_policy_from_virtual_node_yaml_str(
             # mounts
             mounts = copy.deepcopy(config.DEFAULT_MOUNTS_VIRTUAL_NODE)
             volumes = case_insensitive_dict_get(spec, "volumes") or []
-            configmap_volume_names = [
-                case_insensitive_dict_get(volume, "name") if "configmap" in volume else None for volume in volumes
-            ]
+
+            # track volumes that have default read-only behavior (configmap, secret, downward API)
+            read_only_volume_types = {}
+            for volume in volumes:
+                volume_name = case_insensitive_dict_get(volume, "name")
+                # Only handle volume types that are read-only by default
+                if "configMap" in volume:
```

Comment:
this logic can be simplified:
# have this outside of the 'for volume in volumes'
ro_volumes = ["configMap", "secret", "downwardAPI"]
for v in ro_volumes:
   if v in volume:
      read_only_volume_types[v] = v

Reply:
Actually, why is this a dictionary anyway?

Reply:
ah all good points, thanks! we don't need to distinguish between them. just pushed a more streamlined version